index.js

```diff
@@ -130,12 +27,42 @@ exports.lintFiles = function (patterns, opts) {
 			return ext === '.js' || ext === '.jsx';
 		});
 
-		var engine = new eslint.CLIEngine(opts._config);
-
-		return engine.executeOnFiles(paths);
+		if (!(opts.overrides && opts.overrides.length)) {
+			return runEslint(paths, opts);
+		}
+
+		var overrides = opts.overrides;
+		delete opts.overrides;
+
+		var grouped = optionsManager.groupConfigs(paths, opts, overrides);
+
+		return Promise.all(grouped.map(function (data) {
```

Comment:
Why use `Promise.all` here? `runEslint` and `engine.executeOnFiles` are sync functions.